awsshell/utils.py

```diff
@@ -22,6 +26,25 @@ def build_config_file_path(file_name):
     return os.path.join(os.path.expanduser('~'), '.aws', 'shell', file_name)
 
 
+@contextlib.contextmanager
+def temporary_file(mode):
+    """Cross platform temporary file creation.
+
+    This is an alternative to ``tempfile.NamedTemporaryFile`` that
+    also works on windows and avoids the "file being used by
+    another process" error.
+    """
+    tempdir = tempfile.mkdtemp()
+    basename = 'tmpfile-%s' % (uuid.uuid4())
+    full_filename = os.path.join(tempdir, basename)
+    open(full_filename, 'w').close()
```

Comment:
Yep.  The issue would be if this was used with `'r'` the mode would fail.  We could update this to only create the file if if we're not also opening the file via `'w'`